contract/contracts/Vesting.sol

```diff
@@ -0,0 +1,49 @@
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./FanToken.sol";
+
+contract Vesting is Ownable {
+    using SafeMath for uint256;
+
+    mapping(address => bool) public vestingTokens;
+    mapping(address => uint256) public tokensVestingAmount;
+    mapping(address => address) public tokensRecipient;
+    mapping(address => uint256) public tokensVestingStart;
+    mapping(address => uint256) public tokensVestingEnd;
+    mapping(address => uint256) public tokensLastUpdate;
+
+    function addVesting(
+        address token,
+        address recipient,
+        uint256 vestingAmount,
+        uint256 vestingStart,
+        uint256 vestingEnd
+    ) external onlyOwner {
```

Comment:
If you call this function from the TokenFactory, the owner of this function will be the wallet address where you deployed VestingContract because the TokenFactory doesn't have the function to deploy VestingContract.

I think that TokenFactory should be made to be the owner.

Reply:
Yes, I wrote that scenario on unit test(actual deploy process is not considered though...)